les/clientpool.go

```diff
@@ -627,6 +761,37 @@ func (db *nodeDB) delPB(id enode.ID) {
 	db.pcache.Remove(string(key))
 }
 
+// getPosBalanceIDs returns a lexicographically ordered list of IDs of accounts
+// with a positive balance
+func (db *nodeDB) getPosBalanceIDs(start, stop enode.ID, maxCount int) (result []enode.ID) {
+	if maxCount <= 0 {
+		return
+	}
+	it := db.db.NewIteratorWithStart(db.key(start.Bytes(), false))
+	defer it.Release()
+	for i := len(stop[:]) - 1; i >= 0; i-- {
```

Comment:
It seems unnecessarily complicated.

Why not extract the id from `iter.Key` every time and compare with `stop` enode.ID?

Anyway, we need to extract the ID to return.

Reply:
Because NewIteratorWithStart is not NewIteratorWithPrefix so this iterator can run out of the balance entry range. Comparing the entire key ensures that we stop as soon as this happens.
Btw is there any way to iterate only in a prefix range but also start from a given key? That would allow us to simplify this routine but I don't know about any easy way to do that.